internal/services/pam/pam_test.go

```diff
@@ -0,0 +1,606 @@
+			require.Equal(t, want, got, "AvailableBrokers returned unexpected brokers")
+		})
+	}
+}
+
+func TestSetDefaultBrokerForUser(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		username string
+		noBroker bool
+
+		wantErr bool
+	}{
+		"Set default broker for existing user": {username: "user"},
+
+		"Error when username is empty":     {wantErr: true},
+		"Error when broker does not exist": {username: "user", noBroker: true, wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			client := startClient(t, filepath.Join(fixturesPath, "mixed_brokers"))
+
+			abResp, err := client.AvailableBrokers(context.Background(), &authd.Empty{})
+			require.NoError(t, err, "Setup: failed to get available brokers for tests")
+
+			//nolint:gosec // This just to vary the selected broker in the tests, we don't care about randomness safety.
+			wantID := abResp.GetBrokersInfos()[rand.Intn(len(abResp.GetBrokersInfos()))].GetId()
+			if tc.noBroker {
+				wantID = "does not exist"
+			}
+
+			sdbfuReq := &authd.SDBFURequest{
+				BrokerId: wantID,
+				Username: tc.username,
+			}
+			_, err = client.SetDefaultBrokerForUser(context.Background(), sdbfuReq)
+			if tc.wantErr {
+				require.Error(t, err, "SetDefaultBrokerForUser should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "SetDefaultBrokerForUser should not return an error, but did")
+
+			gotResp, _ := client.GetPreviousBroker(context.Background(), &authd.GPBRequest{Username: tc.username})
+			require.Equal(t, wantID, gotResp.GetPreviousBroker(), "SetDefaultBrokerForUser did not set the correct broker for the user")
+		})
+	}
+}
+
+func TestGetPreviousBroker(t *testing.T) {
+	t.Parallel()
+
+	client := startClient(t, filepath.Join(fixturesPath, "mixed_brokers"))
+
+	_, err := client.SetDefaultBrokerForUser(context.Background(), &authd.SDBFURequest{
+		BrokerId: "local",
+		Username: "user",
+	})
+	require.NoError(t, err, "Setup: could not set default broker for user for tests")
+
+	gotResp, _ := client.GetPreviousBroker(context.Background(), &authd.GPBRequest{Username: "user"})
+	require.Equal(t, "local", gotResp.GetPreviousBroker(), "GetPreviousBroker did not return the correct broker")
+
+	gotResp, _ = client.GetPreviousBroker(context.Background(), &authd.GPBRequest{Username: "does not have a broker"})
+	require.Empty(t, gotResp.GetPreviousBroker(), "GetPreviousBroker should return nil when the user has no broker assigned")
+}
+
+func TestSelectBroker(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		mockResponse string
+		brokerID     string
+		username     string
+
+		wantErr bool
+	}{
+		"Successfully select a broker and creates the session": {username: "success"},
+
+		"Error when username is empty":                    {wantErr: true},
+		"Error when brokerID is empty":                    {username: "empty broker", brokerID: "empty", wantErr: true},
```

Comment:
In adsys and other projects, we use "-" to signify empty. Let’s stick to the same convention